config/application.rb

```diff
@@ -20,6 +23,43 @@ class Application < Rails::Application
       end
     end
 
+    config.after_initialize do
+      # We need this here because the test env file loads before the Spree engine is loaded
+      Spree::Core::Engine.routes.default_url_options[:host] = 'test.host' if Rails.env = 'test'
```

Comment:
I'd really like to be able to have this in test.rb but when test.rb runs Spree::Core is not initialized yet, I think because config/initializers/spree.rb has not run yet. This after_initialize here works well but this not not nice:
`if Rails.env = 'test'`